diff --git a/dist/core.cjs b/dist/core.cjs
index c6ddaee813c9b37aeeedb8be15ccc55353869afd..9149c108c2753ce4eb081c9ae619b92650beb80f 100644
--- a/dist/core.cjs
+++ b/dist/core.cjs
@@ -285,8 +285,8 @@ ${unknownFlags.map((i) => `@${i.name}`).join(", ")}`,
               includeCompletionsForModuleExports: false
             });
             completions = result?.entries ?? [];
-            if (completions[0].replacementSpan?.length) {
-              prefix = completions[0]?.replacementSpan && code.slice(
+            if (completions[0]?.replacementSpan?.length) {
+              prefix = code.slice(
                 completions[0].replacementSpan.start,
                 target
               ) || prefix;
diff --git a/dist/core.mjs b/dist/core.mjs
index c052c9e9e9c16e0c3e0c3040c3464e42d3efd21e..5eb54c9db2ba75df0a8406ccb470952a4500d1a2 100644
--- a/dist/core.mjs
+++ b/dist/core.mjs
@@ -284,8 +284,8 @@ ${unknownFlags.map((i) => `@${i.name}`).join(", ")}`,
               includeCompletionsForModuleExports: false
             });
             completions = result?.entries ?? [];
-            if (completions[0].replacementSpan?.length) {
-              prefix = completions[0]?.replacementSpan && code.slice(
+            if (completions[0]?.replacementSpan?.length) {
+              prefix = code.slice(
                 completions[0].replacementSpan.start,
                 target
               ) || prefix;
